File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort();
    let cnt = 0;
    let left = 0;
    let right = people.length - 1;
    
    while(left < right) {
        if(people[left] + people[right] > limit) {
            right--;
        } else {
            cnt++;
            left++;
            right--;
        }
    }
    
    answer += cnt;
    answer += people.length - 2 * cnt;
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end());
    
    int cnt = 0;
    int left = 0;
    int right = people.size()-1;
    while(left < right) {
        if(people[left] + people[right] > limit) {
            right--;
        } else {
            cnt++;
            left++;
            right--;
        }
    }
    answer += cnt;
    answer += people.size() - 2 * cnt;
    return answer;
}